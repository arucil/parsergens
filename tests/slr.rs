#![feature(proc_macro_hygiene)]

use insta::assert_debug_snapshot;

parsergens::parsergen!(mod parser: SLR("fixtures/expr_eval.pg"));

#[test]
fn simple() {
  let tokens = parser::with_input(r"(3.2 * 51 + 1) / 20     ")
    .parse_expr();
  assert_debug_snapshot!(tokens);
}

parsergens::parsergen!(mod parser_with_state: SLR("fixtures/expr_state.pg"));

#[test]
fn state() {
  let tokens = parser_with_state::with_input(r"(3.2 * 51 + 1) / 20     ")
    .parse_expr();
  assert_debug_snapshot!(tokens);
}