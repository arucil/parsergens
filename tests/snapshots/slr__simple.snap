---
source: tests/slr.rs
expression: tokens
---
Ok(
    Ok(
        8.21,
    ),
)
